packages/teleport/src/Main/Main.tsx

```diff
@@ -26,10 +26,12 @@ import SideNav from 'teleport/SideNav';
 import TopBar from 'teleport/TopBar';
 import getFeatures from 'teleport/features';
 import useMain, { State } from './useMain';
+import useTeleport from 'teleport/useTeleport';
 
 export default function Container() {
   const [features] = React.useState(() => getFeatures());
-  const state = useMain(features);
+  const ctx = useTeleport();
```

Comment:
is it only for consistency reasons you prefer to inject the ctx this way or there is something else?

Reply:
so i can pass in enterprise context for enterprise Main

Reply:
> so i can pass in enterprise context for enterprise Main

But all the components will continue referencing the OSS context because this is what remains in the ContextProvider. 
For E, you need to initialize TeleportContextProvider with the E version.


```tsx
export default function TeleportE({ history }) {
  const [ctx] = React.useState( () => new TeleportE() );
  return (
    <Teleport history={history} ctx={ctx}>
      <AccessStrategy>
        <LicenseEnforcer />
```

Then in every place where we have `useTeleport()` the E version will be returned. 

https://reactjs.org/docs/context.html